src/mount.c

```diff
@@ -103,10 +186,16 @@ pu_umount_all(const gchar *device,
 gboolean
 pu_device_mounted(const gchar *device)
 {
-    g_autofree gchar *cmd = g_strdup("mount");
-    g_autofree gchar *output = NULL;
+    g_autoptr(GList) mounted = NULL;
+    g_autoptr(GError) error = NULL;
+
+    g_return_val_if_fail(g_strcmp0(device, "") > 0, FALSE);
+
+    if (!pu_find_mounted(device, &mounted, &error))
+        return TRUE;
 
-    g_spawn_command_line_sync(cmd, &output, NULL, NULL, NULL);
+    if (g_list_length(mounted))
+        return TRUE;
```

Comment:
I think this should check the case when there are no items in `mounted`, meaning the negative (opposite) case. If all things are fine, the function should proceed to the final `return` statement with `TRUE`.

Reply:
Same here, `pu_device_mounted()` is only used to find out if the device is not mounted so `FALSE` is the positive case.
I also changed this check to a `NULL` check as it is faster.